File: Documentation/4.0/Samples/java/src/test/java/net/ravendb/Indexes/Boosting.java
package net.ravendb.Indexes;

import net.ravendb.client.documents.DocumentStore;
import net.ravendb.client.documents.IDocumentStore;
import net.ravendb.client.documents.indexes.AbstractIndexCreationTask;
import net.ravendb.client.documents.indexes.IndexDefinition;
import net.ravendb.client.documents.operations.indexes.PutIndexesOperation;
import net.ravendb.client.documents.session.IDocumentSession;

import java.util.Collections;
import java.util.List;

public class Boosting {

    //region boosting_2
    public class Employees_ByFirstAndLastName extends AbstractIndexCreationTask {
        public Employees_ByFirstAndLastName() {
            map = "docs.Employees.Select(employee => new {" +
                "    FirstName = employee.FirstName.Boost(10)," +
                "    LastName = employee.LastName" +
                "})";
        }
    }
    //endregion

    private static class Employee {

    }

    public Boosting() {
        try (IDocumentStore store = new DocumentStore()) {
            try (IDocumentSession session = store.openSession()) {
                //region boosting_3
                // employees with 'firstName' equal to 'Bob'
                // will be higher in results
                // than the ones with 'lastName' match
                List<Employee> results = session.query(Employee.class, Employees_ByFirstAndLastName.class)
                    .whereEquals("FirstName", "Bob")
                    .whereEquals("LastName", "Bob")
                    .toList();
                //endregion
            }

            //region boosting_4
            IndexDefinition indexDefinition = new IndexDefinition();
            indexDefinition.setName("Employees/ByFirstAndLastName");
            indexDefinition.setMaps(Collections.singleton(
                "docs.Employees.Select(employee => new {" +
                "    FirstName = employee.FirstName.Boost(10)," +
                "    LastName = employee.LastName" +
                "})"));

            store.maintenance().send(new PutIndexesOperation(indexDefinition));
            //endregion
        }
    }
}

File: Documentation/4.0/Samples/nodejs/indexes/boosting.js
import { 
    DocumentStore, AbstractIndexCreationTask, IndexDefinition, PutIndexesOperation
} from "ravendb";

const store = new DocumentStore();

//region boosting_2
class Employees_ByFirstAndLastName extends AbstractIndexCreationTask {
    constructor() {
        super();
        this.map = "docs.Employees.Select(employee => new {" +
            "    firstName = employee.FirstName.Boost(10)," +
            "    lastName = employee.LastName" +
            "})";
    }
}
//endregion

class Employee { }

async function boosting() {
    
    {
        const session = store.openSession();

        //region boosting_3
        // employees with 'FirstName' equal to 'Bob'
        // will be higher in results
        // than the ones with 'LastName' match
        const results = await session.query({ indexName: "Employees/ByFirstAndLastName" })
            .whereEquals("FirstName", "Bob")
            .whereEquals("LastName", "Bob")
            .all();
        //endregion
    }

    //region boosting_4
    const indexDefinition = new IndexDefinition();
    indexDefinition.name = "Employees/ByFirstAndLastName";
    indexDefinition.maps = new Set([ 
        "docs.Employees.Select(employee => new {" +
        "    FirstName = employee.FirstName.Boost(10)," +
        "    LastName = employee.LastName" +
        "})" 
    ]);

    await store.maintenance.send(new PutIndexesOperation(indexDefinition));
    //endregion
}
